app/src/main/java/com/mk/tasky/agenda/detail/reminder/presentation/DetailReminderViewModel.kt

```diff
@@ -29,6 +34,63 @@ class DetailReminderViewModel @Inject constructor(
                 time = LocalDateTime.parse(it).toLocalTime()
             )
         }
+
+        val itemId = savedStateHandle.get<String>("id")
+        if (itemId != null) {
+            viewModelScope.launch {
+                val reminder = reminderUseCases.getReminder(itemId)
+                state = state.copy(
+                    id = itemId,
+                    title = reminder.title,
+                    description = reminder.description,
+                    date = reminder.dateTime.toLocalDate(),
+                    time = reminder.dateTime.toLocalTime(),
+                    reminder = calculateRemindAtTime(reminder)
+                )
+            }
+            savedStateHandle.get<String>("action")?.let {
+                when (HomeItemOptions.from(it)) {
+                    HomeItemOptions.EDIT -> {
+                        state = state.copy(
+                            isEditing = true
+                        )
+                    }
+                    HomeItemOptions.OPEN -> {
+                        state = state.copy(
+                            isEditing = false
+                        )
+                    }
+                    HomeItemOptions.DELETE -> {
+                        viewModelScope.launch(NonCancellable) {
+                            reminderUseCases.deleteReminder(itemId)
+                            state = state.copy(
```

Comment:
Here `NonCancellable` is not needed since `deleteReminder` will suspend as long as it's deleted and only then you'll navigate away.
If I remember correctly at the other part in your code where I commented on that, you updated the state outside of the coroutine, right? Don't fully remember

Reply:
Oh you are right. Fixed!